frappe/desk/doctype/todo/todo.js

```diff
@@ -1,40 +1,90 @@
 // bind events
 
 frappe.ui.form.on("ToDo", {
-	onload: function(frm) {
-		frm.set_query("reference_type", function(txt) {
+	onload: function (frm) {
+		frm.set_query("reference_type", function (txt) {
 			return {
 				"filters": {
 					"issingle": 0,
 				}
 			};
 		});
 	},
-	refresh: function(frm) {
-		if(frm.doc.reference_type && frm.doc.reference_name) {
-			frm.add_custom_button(__(frm.doc.reference_name), function() {
+
+	validate: function (frm) {
+		if (frm.doc.is_repeating && frm.doc.__islocal) {
+			frm.trigger("new_auto_repeat_prompt");
+		}
+	},
+
+	refresh: function (frm) {
+		if (frm.doc.reference_type && frm.doc.reference_name) {
+			frm.add_custom_button(__(frm.doc.reference_name), function () {
 				frappe.set_route("Form", frm.doc.reference_type, frm.doc.reference_name);
 			});
 		}
 
 		if (!frm.doc.__islocal) {
-			if(frm.doc.status!=="Closed") {
-				frm.add_custom_button(__("Close"), function() {
+			if (frm.doc.status !== "Closed") {
+				frm.add_custom_button(__("Close"), function () {
 					frm.set_value("status", "Closed");
-					frm.save(null, function() {
+					frm.save(null, function () {
 						// back to list
 						frappe.set_route("List", "ToDo");
 					});
 				}, "fa fa-check", "btn-success");
 			} else {
-				frm.add_custom_button(__("Reopen"), function() {
+				frm.add_custom_button(__("Reopen"), function () {
 					frm.set_value("status", "Open");
 					frm.save();
 				}, null, "btn-default");
 			}
-			frm.add_custom_button(__("New"), function() {
+			frm.add_custom_button(__("New"), function () {
 				frappe.new_doc("ToDo")
 			}, null, "btn-default");
 		}
+	},
+
+	new_auto_repeat_prompt: function (frm) {
```

Comment:
we can add a generic `allow_auto_repeat` and `auto_repeat_date` property on DocType so that we can make any document ready for auto-repeat. This can be then extended to other objects like Task, etc.

The code will be very similar

Reply:
Hello, I have confusion in `auto_repeat_date`. Is this a starting date or an end date?

Reply:
Are the properties to be made only on `ToDo` or the general DocType?

Reply:
general DocType

auto-repeat should be for start date (ideally!)

Move this to a common file

Reply:
Thank you for pointing out! :grinning:

Reply:
Hello,
I have made some changes below. I didn't add `auto_repeat_date`, I feel like it is redundant. Since Auto Repeat requires any DocType to have `auto_repeat` field.